app/actions/local/post.ts

```diff
@@ -132,6 +132,29 @@ export const selectAttachmentMenuAction = (serverUrl: string, postId: string, ac
     return postActionWithCookie(serverUrl, postId, actionId, '', selectedOption);
 };
 
+export const markPostAsDeleted = async (serverUrl: string, post: Post) => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const dbPost = await queryPostById(operator.database, post.id);
+    if (!dbPost) {
+        return {};
+    }
+
+    dbPost.prepareUpdate((p) => {
+        p.deleteAt = Date.now();
+        p.message = '';
+        p.metadata = null;
+        p.props = undefined;
```

Comment:
I'm not sure just yet, but we may need to change this to have a prop for a `deleted` state

Reply:
Wouldn't that be redundant with `deleteAt > 0`?

Reply:
So, this is the thing. UX wise there is a (message deleted) post being shown when some other user deletes a post a in the channel and tapping on the post removes it from the post list.
So I wanted to add the comment here just to create awareness about this and start thinking into how we are going to deal with it

Reply:
The way I would deal with it is:
- When a post deleted event comes from the websocket event, we "sanitize" and add the `deleteAt` value.
- When a "(message deleted)" post is clicked, we remove the post from the database.
- The list of posts will show any `deleteAt > 0` post as "(message deleted)".

The only thing that would require a bit more of thought is synchronization.
- I have post A on my database.
- My websocket is not connected when post A is deleted, so I don't receive the websocket message.
- I connect and fetch all the posts. Post A is not in the list of posts.
- Should we mark post A as deleted, or should we remove it from the database?

In this scenario, I would mark it as deleted, but I worry what happens with things like pagination, and of course, the security implications of remnant data in the database with "deleted information".